framework/src/modules/chain/dpos/delegates_info.js

```diff
@@ -85,6 +85,16 @@ class DelegatesInfo {
 
 		// Perform updates that only happens in the end of the round
 		if (this._isLastBlockOfTheRound(block)) {
+			if (undo) {
```

Comment:
Can you also remove the `@todo` mention and add a small description for the `_update` function as it's a generic name. (out of scope+-)